<gazebo version="1.0">
  <world name="world_1">
    <scene>
      <ambient rgba="0.5 0.0 0.0 0.5"/>
      <background rgba="0.0 0.0 0.5 0.5">
        <sky mateiral="Gazebo/Sky"/>
      </background>
      <shadows enabled="true" rgba="0.0 0.0 0.5 0.5" type="texture_gpu"/>
      <fog rgba="0.0 0.0 0.5 0.5" type="linear" start="2.0" end="10.0" density="0.5"/>
    </scene>

    <physics type="ode">
      <ode>
        <max_contacts>250</max_contacts>
        <solver type="quick" dt="0.001" iters="20" sor="1.3"/>
        <constraints cfm="0" erp="0.2" contact_max_correctiong_vel="100.0" contact_surface_layer="0.0"/>
        <gravity xyz="0.0 0.0 -9.81"/>
      </ode>
    </physics>

    <model name="model_1" static="false">

      <origin pose="0.0 0.0 1.0 0.0 0.0 0.0"/>

      <link name="link_1" self_collide="true" gravity="false" kinematic="false">
        <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
        <inertial mass="10.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <damping linear="0.0" angular="0.0" />
          <inertia ixx="1.0" ixy="0.0" ixz="0.0"
                   iyy="1.0" iyz="0.0"
                   izz="1.0"/>
        </inertial>

        <visual cast_shadows="true" name="visual_box" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <box size="1.0 1.0 2.0"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
        </visual>

        <visual cast_shadows="true" name="visual_cylinder" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <cylinder radius="1.0" length="2.0"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
        </visual>

        <visual cast_shadows="true" name="visual_sphere" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <sphere radius="0.5"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
        </visual>

        <visual cast_shadows="true" name="visual_plane" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <plane normal="0.0 0.0 1.0"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
        </visual>

        <visual cast_shadows="true" name="visual_image" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <image filename="test.png" scale="2.0" threshold="10" height="3.0" granularity="2"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
        </visual>

        <visual cast_shadows="true" name="visual_heightmap" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <heightmap filename="test.png" size="1.0 1.0 2.0" offset="0.0 0.0 0.0"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
        </visual>

        <visual cast_shadows="true" name="visual_mesh" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <mesh filename="test.dae" scale="1.0 1.0 2.0"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
        </visual>

        <collision name="collision_box" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <box size="1.0 1.0 2.0"/>
          </geometry>
          <surface>
            <max_contacts>250</max_contacts>
            <friction>
              <ode mu="0.5" mu2="0.2" fdir1="1.0 0 0" slip1="0" slip2="0"/>
            </friction>
            <bounce restitution_coefficient="0" threshold="1000000.0"/>
            <contact>
              <ode soft_cfm="0" soft_erp="0.2" kp="1e15" kd="1e13" max_vel="100.0" min_depth="0.0001"/>
            </contact>
          </surface>
        </collision>

        <collision name="collision_cylinder" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <cylinder radius="1.0" length="2.0"/>
          </geometry>
          <surface>
            <friction>
              <ode mu="0.5" mu2="0.2" fdir1="1.0 0 0" slip1="0" slip2="0"/>
            </friction>
            <bounce restitution_coefficient="0" threshold="1000000.0"/>
            <contact>
              <ode soft_cfm="0" soft_erp="0.2" kp="1e15" kd="1e13" max_vel="100.0" min_depth="0.0001"/>
            </contact>
          </surface>
        </collision>

        <collision name="collision_sphere" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <sphere radius="0.5"/>
          </geometry>
          <material script="Gazebo/Green" normal_map="normal_map.png">
            <color rgba="0.03 0.5 0.5"/>
          </material>
          <surface>
            <friction>
              <ode mu="0.5" mu2="0.2" fdir1="1.0 0 0" slip1="0" slip2="0"/>
            </friction>
            <bounce restitution_coefficient="0" threshold="1000000.0"/>
            <contact>
              <ode soft_cfm="0" soft_erp="0.2" kp="1e15" kd="1e13" max_vel="100.0" min_depth="0.0001"/>
            </contact>
          </surface>
        </collision>

        <collision name="collision_plane" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <plane normal="0.0 0.0 1.0"/>
          </geometry>
          <surface>
            <friction>
              <ode mu="0.5" mu2="0.2" fdir1="1.0 0 0" slip1="0" slip2="0"/>
            </friction>
            <bounce restitution_coefficient="0" threshold="1000000.0"/>
            <contact>
              <ode soft_cfm="0" soft_erp="0.2" kp="1e15" kd="1e13" max_vel="100.0" min_depth="0.0001"/>
            </contact>
          </surface>
        </collision>

        <collision name="collision_image" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <image filename="test.png" scale="2.0" threshold="10" height="3.0" granularity="2"/>
          </geometry>
          <surface>
            <friction>
              <ode mu="0.5" mu2="0.2" fdir1="1.0 0 0" slip1="0" slip2="0"/>
            </friction>
            <bounce restitution_coefficient="0" threshold="1000000.0"/>
            <contact>
              <ode soft_cfm="0" soft_erp="0.2" kp="1e15" kd="1e13" max_vel="100.0" min_depth="0.0001"/>
            </contact>
          </surface>
        </collision>

        <collision name="collision_heightmap" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <heightmap filename="test.png" size="1.0 1.0 2.0" offset="0.0 0.0 0.0"/>
          </geometry>
          <surface>
            <friction>
              <ode mu="0.5" mu2="0.2" fdir1="1.0 0 0" slip1="0" slip2="0"/>
            </friction>
            <bounce restitution_coefficient="0" threshold="1000000.0"/>
            <contact>
              <ode soft_cfm="0" soft_erp="0.2" kp="1e15" kd="1e13" max_vel="100.0" min_depth="0.0001"/>
            </contact>
          </surface>
        </collision>

        <collision name="collision_mesh" laser_retro="100.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <geometry>
            <mesh filename="test.dae" scale="1.0 1.0 2.0"/>
          </geometry>
          <surface>
            <friction>
              <ode mu="0.5" mu2="0.2" fdir1="1.0 0 0" slip1="0" slip2="0"/>
            </friction>
            <bounce restitution_coefficient="0" threshold="1000000.0"/>
            <contact>
              <ode soft_cfm="0" soft_erp="0.2" kp="1e15" kd="1e13" max_vel="100.0" min_depth="0.0001"/>
            </contact>
          </surface>
        </collision>


        <sensor name="sensor_camera" always_on="true" update_rate="2.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <camera>
            <horizontal_fov angle="1.57"/>
            <image width="2560" height="1600" format="BAYER_BGGR8"/>
            <clip near="0.01" far="100.0"/>
            <save enable="false" path="/tmp"/>
          </camera>
        </sensor>

        <sensor name="sensor_ray" always_on="true" update_rate="2.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <ray>
            <scan display="true">
              <horizontal sampes="500" />
              <image width="2560" height="1600" format="BAYER_BGGR8"/>
              <clip near="0.01" far="100.0"/>
              <save enable="false" path="/tmp"/>
            </scan>
          </ray>
        </sensor>

        <sensor name="sensor_camera" always_on="true" update_rate="2.0">
          <origin pose="0.0 0.0 0.0 0.0 0.0 0.0"/>
          <camera>
            <horizontal_fov angle="1.57"/>
            <image width="2560" height="1600" format="BAYER_BGGR8"/>
            <clip near="0.01" far="100.0"/>
            <save enable="false" path="/tmp"/>
          </camera>
        </sensor>

      </link>

      <link name="link_2"/>

      <joint name="model_joint_1">
      </joint>

    </model>

    <light name="light_1">
    </light>

    <joint name="world_joint_1">
    </joint>

    <plugin name="plugin_1">
    </plugin>

  </world>
</gazebo>
